### PR TITLE
Docs: add more delta updates info and rockset-specific caveats

### PR DESCRIPTION
**Pull request by oliviamiannone**
**Description:**

In light of https://github.com/estuary/connectors/issues/115, the Rockset connector can handle TOASTed values without issue, but there are important considerations regarding delta updates and reduction strategies that users should be aware of.

**Documentation links affected:**
na

**Notes for reviewers:**

The usual - needs a fact-check as some assumptions were involved. One specific question noted inline.

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/399)

### CODE REVIEW

File: site/docs/reference/Connectors/materialization-connectors/Rockset.md [hunk 1]
```diff
@@ -55,6 +55,22 @@ materializations:
       	collection: ${table_name}
     source: ${tenant}/${source_collection}
 ```
+
+## Delta updates and reduction strategies
+
+The Rockset connector operates only in [delta updates](../../../concepts/materialization.md#delta-updates) mode.
+This means that Rockset merges documents by key when they arrive, using semantics that are somewhat different than Flow's.
```
Comment (psFried) [inline site/docs/reference/Connectors/materialization-connectors/Rockset.md:62]: Here's my take on this:

> Rockset will merge documents having the same key, as defined by the Flow collection. The documents will be merged according to semantics of [RFC 7396 - JSON merge](https://datatracker.ietf.org/doc/html/rfc7396). This is different from how Flow reduces documents, and Rockset's merge will _not_ use any reduction annotations from your Flow schema. If your Flow JSON schema has a top-level `merge` reduction strategy and all other fields have the default reduction strategy of `lastWriteWins`, then the results will be equivalent. If your Flow collection's JSON schema uses any other reduction stategies, then the view that's materialized to Rockset will differ from the view of the same collection that's materialized to an RDBMS such as Postgres.

File: site/docs/reference/Connectors/materialization-connectors/Rockset.md [hunk 2]
```diff
@@ -55,6 +55,22 @@ materializations:
...
+(FOR REVIEW: include a brief description of the difference here? I don't quite understand the distinction but seems worthwhile to cover in brief)
+
+In most cases, this difference won't affect your data. However, using certain
```
Comment (psFried) [inline site/docs/reference/Connectors/materialization-connectors/Rockset.md:65]: > In most cases, this difference won't affect your data

This isn't going to age well, as I expect that reduction annotations will become much more commonly used than they are now. I'm a little torn about whether to leave it for now, but just FYI

**Review by psFried — CHANGES_REQUESTED** [review]
I think this helps a lot. I gave a suggestion for `FOR REVIEW` paragraph, but everything else looks great

File: site/docs/reference/Connectors/materialization-connectors/Rockset.md (hunk 1 above)
Comment (willdonnelly) [reply site/docs/reference/Connectors/materialization-connectors/Rockset.md:62]: s/will differ/may differ/ perhaps?

**Pull request merged**